src/process.ml

```diff
@@ -281,8 +281,10 @@ let run_internal ?dir ?(stdout_to=Terminal) ?(stderr_to=Terminal) ~env ~purpose
   Option.iter to_close ~f:Unix.close;
   close_std_output close_stdout;
   close_std_output close_stderr;
+  let stats_event = Stats.Catapult.on_process_start ~program:prog_str ~args in
```

Comment:
Do we want to do these operations conditionally? Only in the case where the tracing is enabled for example. Otherwise, they might slow things down. Perhaps:

```
Stats.Catapult.with_trace (fun () ->
  ..
)
```

And `with_trace` will just run its argument when the tracing isn't enabled.

Reply:
When this is disabled, we're just allocating a record with the event data. That shouldn't be very expensive compared to the associated fork.

Reply:
I see. Well even if it's not needed for performance, I think using a bracket like pattern is nicer. You don't have to remember to end things, and it can also give us exception safety for free.

Reply:
I'm not sure how to bracket it here though, since the pattern is slightly different:

```ocaml
let event = start () in
mx >>| fun x ->
end event;
rest
```

Reply:
Is the monad, the issue? You could always do:

```
val with_trace : program:string -> args:string list -> (unit -> 'a Fiber.t)
```

Where we run the callback, `>>|` on the result to call `on_end_process`.

In any case, such a function couldn't go in the Catapult library as it would add a dependence on Fiber.

Reply:
That worked!